jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
I gave this some thought at the time and I couldn't find a way around the client being able to supply the requested `sys.path` to Jedi in public API. Now, to my liking, full `sys_path` is too long for a proper interface parameter and most of the time it will refer to one or more virtualenvs, so probably specifying virtualenv paths will look better. For example, we could add `venvs=[path1, path2]` to use those venvs and `extra_sys_path=[path1, path2]` just in case someone wants, say, to additionally use some source code checkouts without installing them into the said venvs.

If this API addition goes through, I'm not sure if it is worth keeping the `VIRTUAL_ENV` logic: 
it boils down to having both `VIRTUAL_ENV`'s and jedi's `sys.path` in one interpreter and this is better accomplished the other way around. I mean, that logic starts with jedi on `sys.path` and then tries to add venv path afterwards, but as long as jedi has no external dependencies, we can just start up `PATH/TO/VENV/bin/python PATH/TO/JEDI/portable.py` with `portable.py` adding its location to `sys.path` and then running jedi as usual. That way it will have more binary compatibility with code in the venv.